File: Object-Oriented Programming/Projects/EncapsulationAndPlymorphism/Shapes/Circle.cs
using System;

public class Circle : IShape
{
    private double radius;

    public Circle(double radius)
    {
        this.Radius = radius;
    }

    public double Radius
    {
        get
        {
            return this.radius;
        }

        set
        {
            Checkers.CheckSide(value);

            this.radius = value;
        }
    }

    public double CalculateArea()
    {
        double area = Math.PI * Math.Pow(this.Radius, 2);

        return area;
    }

    public double CalculatePerimeter()
    {
        double perimeter = 2 * Math.PI * this.Radius;

        return perimeter;
    }
}

File: Object-Oriented Programming/Projects/OOP In Java/Problem1_Geometry/src/geometry/shape/planeShape/Circle.java
package geometry.shape.planeShape;

import geometry.Vertex;

import java.util.Arrays;
import java.util.List;

public class Circle extends PlaneShape {
	private double radius;
	
	public Circle(List<Vertex> vertices, double radius) {
		super(vertices);
		setRadius(radius);
	}
	
	public Circle(Vertex centreVertex, double radius) {		
		this(Arrays.asList(centreVertex), radius);		
	}

	public double getRadius() {
		return radius;
	}

	public void setRadius(double radius) {
		if (radius <=0) {
			throw new IllegalArgumentException("Radius has to be more than 0");
		}
		this.radius = radius;
	}
	
	@Override
	public void setVertices(List<Vertex> upperLeftVertex) {
		if(upperLeftVertex.size() != 1) {
			throw new IllegalArgumentException("Circle constructor can take only one vertex");
		}
		
		super.setVertices(upperLeftVertex);
	}
	
	@Override
	public double getArea() {
		double area = Math.PI*this.radius*this.radius;
		return area;
	}

	@Override
	public double getPerimeter() {
		double perimeter = 2*Math.PI*this.radius;
		return perimeter;
	}

	@Override
	public String toString() {
		String output = "Plane shape: circle\n";
		output += "Radius: " + this.getRadius() + "\n";
		output += "Circle centre: \n";
		output += super.toString();
		return output;
	}
	
	
}
